cfme/fixtures/authentication.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+from time import sleep
+
+import cfme.utils.auth as authutil
+from cfme.utils.appliance import current_appliance
+from cfme.utils.log import logger
+
+
+def get_auth_settings(appliance):
+    """Grab the authentication settings from the UI form, popping the title widget content out"""
+    settings = {'auth_mode': appliance.server.authentication.auth_mode}
+    settings['auth_settings'] = appliance.server.authentication.auth_settings
+    settings['auth_settings'].pop('title')
+    return settings
+
+
+@pytest.fixture(scope='module')
+def amazon_auth_provider():
+    try:
+        return authutil.get_auth_crud('amazon')
+    except KeyError:
+        pytest.skip('amazon auth provider not found in auth_data.auth_providers, skipping test.')
+
+
+@pytest.fixture(scope='module')
+def setup_aws_auth_provider(appliance, amazon_auth_provider):
+    """Configure AWS IAM authentication mode"""
+    appliance.server.authentication.configure(auth_mode='amazon',
+                                              auth_provider=amazon_auth_provider)
+    yield
+
+    current_appliance.server.login_admin()
+    appliance.server.authentication.configure(auth_mode='database')
+
+
+@pytest.fixture(scope='function')
+def auth_provider(prov_key):
+        return authutil.get_auth_crud(prov_key)
+
+
+@pytest.fixture(scope='function')
+def auth_user_data(auth_provider, user_type):
+    try:
+        data = [user
+                for user in auth_provider.user_data
+                if user_type in user.user_types]
```

Comment:
**QUESTION:** If they gave us an invalid user_type wouldn't it be better to just fail.

Reply:
So this isn't about the user_type being right or wrong, its about matching the user_type that is parametrized for the test to users in the yaml.

The yaml contains entries for users, and that user is configured with a certain type. When the test is running against the 'principal' user type, we don't want to include the example user below, since its marked with 'uid' type.

This will return the user data in the case where it exists in the yaml, and skip the test otherwise because our yaml doesn't support that parametrization.

```yaml
test_users:
    -
      username: ldapuser2
      password: Redhat123
      fullname: Ldap User2
      groupname: customgroup1
      providers:
        - freeipa01
      user_types:
        - uid
```

Reply:
**REQUIRED:** There is a comment somewhere in all you said.  Try to figure out the most discrete way of saying that and leave that comment.